### ISSUE TITLE
forward length != reverse length

### ISSUE THREAD

**Issue by donhcd** [unicode-rs/unicode-segmentation#14]
Extremely sorry if this is actually part of the spec, but this doesn't seem quite right. With this code:

```
#[test]
fn it_works() {
    use unicode_segmentation::UnicodeSegmentation;

    let s = "🚁🚦👨‍👧‍👦👩‍👧‍👦";
    println!("forwards:");
    for cluster in s.graphemes(true) {
        println!("{}", cluster);
    }
    println!("backwards:");
    for cluster in s.graphemes(true).rev() {
        println!("{}", cluster);
    }
    assert_eq!(s.graphemes(true).count(), s.graphemes(true).rev().count());
}
```

both with 1.0.1 and master, I get
```
running 1 test
test it_works ... FAILED

failures:

---- it_works stdout ----
        forwards:
🚁
🚦
👨‍👧‍👦
👩‍👧‍👦
backwards:
‍👦
‍👧
👩
‍👦
‍👧
👨
🚦
🚁
thread 'it_works' panicked at 'assertion failed: `(left == right)` (left: `4`, right: `8`)', src/lib.rs:16
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by mbrubeck** [comment]
This is a bug in the reverse iterator, fixed in #17.  Thanks!

**Comment by donhcd** [comment]
awesome, thanks for fixing!
